### ISSUE TITLE
Is there anyway to use my real data (.BVH file) from other mocap (not Xsens)

### ISSUE THREAD

**Issue by hanghoo** [AnyBody/anymocap#36]
I want to know how could use my own test real data from other mocap device(not Xsens). I am not sure whether it is the data saving format problem?
The attachment is my test BVH file.
Hope to hear from you soon!
[BVH file.zip](https://github.com/AnyBody/anymocap/files/2853928/BVH.file.zip)

**Comment by melund** [comment]
Hi @Hoototo. The example model is setup using Xsens and the BVH file exported using the MotionBuilder output format. 

We have created models with many other types of BVH files. It usually requires that you change some things, since the BVH rig can be different. Different names and segments. The BVH structure can be a little difficult to understand but it is possible. 

The biggest problem is that other inertial MoCap system (other than Xsens) have problems with dynamics. They simply dont produce accelerations that make sense dynamically. Xsens is 'state of the art' that way and generally produce nice results.

**Comment by melund** [comment]
@Hoototo. It looks like a file from the  "Perception Neuron" mocap system. Am I right? I think it is definitely possible to create a model with that data. 

I could probably also help you do it. Especially, if you had some file which could be made publicly available so the example could be shared with everyone. 

A file which is relatively static ( not moving too much) is preferable. As I wrote earlier, most inertial MoCap systems get the dynamics wrong. That problem is minimized if motions are slow and not too extreme.

**Comment by hanghoo** [comment]
Thank you! @melund 
I will collect some data tomorrow afternoon and sent it to you.
Thanks a lot.

**Comment by melund** [comment]
No problem. We are going to add the example to our beta AMMR repository. 

Work on the AMMR takes place in a closed repository on https://gitlab.com. It is not on GitHub because our academic collaborators often need to publish their models. So it can be completely public,  but we can give you access. You just need to log in to https://gitlab.com and send me your username 
(you can use your GitHub account when logging into gitlab)

**Comment by hanghoo** [comment]
I have register on gitlab. Username is NickHoo.
I collect 3 kind of data today afternoon. Standing around 1min, Stoop Lifting 1 time, Squat Lifting 1 time.
I export these data, but there are 3 kinds of BVH format. So
 I collect both of them and send to you.
[2.15.2019Standing.zip](https://github.com/AnyBody/anymocap/files/2871276/2.15.2019Standing.zip)
[2.15.2019Squat.zip](https://github.com/AnyBody/anymocap/files/2871278/2.15.2019Squat.zip)
[2.15.2019Stoop.zip](https://github.com/AnyBody/anymocap/files/2871280/2.15.2019Stoop.zip)

**Comment by melund** [comment]
Thanks a lot, @Hoototo. I will try to see what I can do this week. Feel free to ask me how it is progressing :)

**Comment by melund** [comment]
@Hoototo I have added you to the repository on Gitlab: https://gitlab.com/anybody/beta/ammr

**Comment by hanghoo** [comment]
Thank you so much. Sorry for my delay reply.

**Comment by melund** [comment]
@Hoototo I didn't manage to do much about the new BVH example yet. I wanted to use some files from a colleague who did something similar. But he is still on vacation. I will look at it again next week.

**Comment by hanghoo** [comment]
@melund Thank you so much. I am very grateful to your help. After my deadline, I could focus on this issue.

**Comment by melund** [comment]
Hi Nick. So we made a little progress on this. I hope the example will be included in the next version of the AMMR. But I will, of course, share it with you as soon as it is done.

**Comment by hanghoo** [comment]
Hi, Morten. Thanks for your attention. May I ask what kind of tool that you use to open .BVH file?

**Comment by hanghoo** [comment]
In fact, I am not familiar with Anybody. So I need to consult you several questions. Another thing is that I want to confirm to the direction. 
What I need to make Perception Neuron working by using XSens application is to change the .BVH file from up to bottom instead of write code to achieve import Mocap data from bottom to up.
I hope I could have a right direction. So I want to consult you.

**Comment by melund** [comment]
Hi @Hoototo.

I will try to do some of that work for you. But I am running into a small temporary problem with the Perception Neuron BVH file. Right now AnyBody doesn't handle 6 channels for the joints (e.g. the `Xposition Yposition Zposition` )

```
	JOINT Spine
	{
		OFFSET 0 14.307 0
		CHANNELS 6 Xposition Yposition Zposition Yrotation Xrotation Zrotation 
		JOINT Spine1
``` 

I know there is a way to export the data without those data. Could you try to make a file like that for me? That would basically give a file with fixed segment length. A bit similar to the BVH file produced for 3DS Max. Then I will  make the example using that file, and hopefully I can get the next version of AnyBody to support variable segment length.

I should basically look something like this in the exported BVH file:

```
	JOINT Spine
	{
		OFFSET 0 14.307 0
		CHANNELS 3 Yrotation Xrotation Zrotation 
		JOINT Spine1
```

**Comment by melund** [comment]
@Hoototo I don't think you will need to make a 3 channel joint bvh file anyway. I found a way to make a workaround in AnyScript that handles the problem.

 I have created a pull request Perception Neuron MoCap example here: https://gitlab.com/anybody/beta/ammr/merge_requests/356
